File: mobile/src/main/java/com/alexstyl/specialdates/search/NamedayCard.java
package com.alexstyl.specialdates.search;

import com.alexstyl.specialdates.date.Date;
import com.alexstyl.specialdates.events.namedays.NameCelebrations;

public class NamedayCard {

    private NameCelebrations namedays;

    public void setNameday(NameCelebrations nameday) {
        this.namedays = nameday;
    }

    public boolean isAvailable() {
        return namedays != null && namedays.size() > 0;
    }

    public Date getDate(int i) {
        return namedays.getDate(i);
    }

    public void clear() {
        this.namedays = null;
    }

    public String getName() {
        return namedays.getName();
    }

    public int size() {
        return namedays.size();
    }
}

File: android_mobile/src/main/java/com/alexstyl/specialdates/search/NamedayCard.kt
package com.alexstyl.specialdates.search

import com.alexstyl.specialdates.date.Date
import com.alexstyl.specialdates.events.namedays.NameCelebrations

class NamedayCard {

    private var namedays: NameCelebrations? = null

    val isAvailable: Boolean
        get() = namedays != null && namedays!!.dates.isNotEmpty()

    val name: String
        get() = namedays!!.name

    fun setNameday(nameday: NameCelebrations) {
        this.namedays = nameday
    }

    fun getDate(i: Int): Date {
        return namedays!!.dates[i]
    }

    fun clear() {
        this.namedays = null
    }

    fun size(): Int {
        return namedays!!.dates.size
    }
}
